src/test/ui/consts/const-mut-refs/mut_ref_in_final.rs

```diff
@@ -0,0 +1,22 @@
+#![feature(const_mut_refs)]
+#![feature(const_fn)]
+#![feature(raw_ref_op)]
+const NULL: *mut i32 = std::ptr::null_mut();
+const A: *const i32 = &4;
+
+// Must not compile.
+const B: *mut i32 = &mut 4; //~ ERROR untyped pointers are not allowed in constant
```

Comment:
IIRC we considered dropping that "untyped pointers" check... but that would make this code compile. Doing anything bad requires `unsafe`, so that is still okay -- it is just the same as doing `&mut 4 as *mut _` at runtime and copying that raw pointer around too much. So why do you say this *must* not compile?

The thing that really must not compile is `const B: &mut i32`; I assume we already have a test for that?

Reply:
>  So why do you say this must not compile?

I think it is a good extra protection layer that prevents users from accidentally assuming that they can access this mutably via unsafe code. But we could just make that a (clippy?) lint I guess. It feels like a footgun to allow `const B: *mut i32 = &mut 4;` without complaint

Reply:
> I think it is a good extra protection layer that prevents users from accidentally assuming that they can access this mutably via unsafe code. 

Right, I agree with that. But by saying "must", the comment makes it sound like this is a soundness concern, which I think is not. If you code tone down the comment to what you said here, I am happy. :)